units/src/commonMain/kotlin/org/maplibre/spatialk/units/extensions/synthetic.kt

```diff
@@ -58,6 +60,23 @@ public fun Double.toLength(unit: LengthUnit): Length = Length.of(this, unit)
 
 public fun Double.toArea(unit: AreaUnit): Area = Area.of(this, unit)
 
+@OptIn(ExperimentalTypeInference::class)
+@OverloadResolutionByLambdaReturnType
+@JvmName("sumOfLength")
+public inline fun <T> Iterable<T>.sumOf(selector: (T) -> Length): Length =
+    fold(Length.Zero) { acc, t -> acc + selector(t) }
+
+@JvmName("sumLength")
+public fun Iterable<Length>.sum(): Length = fold(Length.Zero) { acc, t -> acc + t }
+
+@OptIn(ExperimentalTypeInference::class)
+@OverloadResolutionByLambdaReturnType
+@JvmName("sumOfArea")
+public inline fun <T> Iterable<T>.sumOf(selector: (T) -> Area): Area =
+    fold(Area.Zero) { acc, t -> acc + selector(t) }
+
+@JvmName("sumArea") public fun Iterable<Area>.sum(): Area = fold(Area.Zero) { acc, t -> acc + t }
+
```

Comment:
extra: so we can do this:

```kt
featureCollection.sumOf { it.geometry!!.length() }
```

so we no longer need overloads for FeatureCollection, GeoJsonObject, etc.

Reply:
to avoid !! one can simply

featureCollection.sumOf { it.geometry?.length() ?: 0 }